application/helpers/frontend_helper.php

```diff
@@ -1865,6 +1865,10 @@ function doAssessment($surveyid, $returndataonly=false)
             $groups=array();
             foreach($fieldmap as $field)
             {
+                // Init Assessment Value
+                $assessmentValue = NULL;
```

Comment:
I just worked on the standard code. Added the event on the same level / code block as the standard assessment process.